application/api/coulumn_manager.go

```
@@ -0,0 +1,113 @@
+var _ ColumnManagerApp = &ColumnManagerAppImpl{}
+
+func (a *ColumnManagerAppImpl) GetAllColumnsWithRelatedTasks(boardId *common.Id) ([]*apimodel.Column, error) {
+	storedColumnsWithRelatedTasks, err := repository.GetColumnRepository().GetAllWithRelatedTasksBy(*boardId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return apidto.NewColumnsFromAggregates(storedColumnsWithRelatedTasks), nil
+}
+
+func (a *ColumnManagerAppImpl) GetColumn(columnId *common.Id) (*apimodel.Column, error) {
+	storedColumn, err := repository.GetColumnRepository().GetBy(*columnId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return apidto.NewColumnFromEntity(storedColumn), nil
+}
+
+func (a *ColumnManagerAppImpl) Create(newColumn *entity.Column) (*apimodel.Column, error) {
+	insertedColumn, err := repository.GetColumnRepository().Insert(newColumn)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return apidto.NewColumnFromEntity(insertedColumn), nil
+}
+
+func (a *ColumnManagerAppImpl) Update(modifiedColumn *entity.Column) (*apimodel.Column, error) {
+	updatedColumn, err := repository.GetColumnRepository().Update(modifiedColumn)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return apidto.NewColumnFromEntity(updatedColumn), nil
+}
+
+func (a *ColumnManagerAppImpl) Rename(columnId common.Id, newName string) (err error) {
+	err = repository.GetColumnRepository().UpdateName(columnId, newName)
+	return
+}
+
+func (a *ColumnManagerAppImpl) ChangePosition(columnId common.Id, newPosition int) (err error) {
+	err = repository.GetColumnRepository().UpdatePosition(columnId, newPosition)
+	return
+}
+
+func (a *ColumnManagerAppImpl) Delete(storedColumnId common.Id) error {
+	parentBoardId, err := repository.GetColumnRepository().GetBoardId(storedColumnId)
+
+	if err != nil {
+		return err
+	}
+
+	isColumnDeletable, err := service.GetBoardService().IsColumnDeletable(parentBoardId, 1)
+
+	if err != nil {
+		return err
+	}
+
+	if !isColumnDeletable {
+		return nil
```

let's return an error if it's not possible to delete a column